src/XPurchase/VirtoCommerce.XPurchase/Schemas/PurchaseSchema.cs

```diff
@@ -643,5 +654,37 @@ private async Task<object> ResolveConnectionAsync(IMediator mediator, IResolveCo
 
             return result;
         }
+
+        private async Task CheckAuthAsync(IResolveFieldContext context, object resource)
+        {
+            var userId = context.GetArgument<string>("userId");
+
+            if (userId == null)
+            {
+                throw new ExecutionError($"argument {nameof(userId)} is null");
+            }
+
+            var signInManager = _signInManagerFactory();
+            var user = await signInManager.UserManager.FindByIdAsync(userId);
+
+            if (user == null)
+            {
+                user = new ApplicationUser()
+                {
+                    Id = userId,
+                    UserName = "Anonymous",
+                };
+                
+                await signInManager.ClaimsFactory.CreateAsync(user);
```

Comment:
Here we already got ClaimPrincilal, no need to signInManager.CreateUserPrincipalAsync later.

Reply:
Fixed